core-bundle/src/Resources/contao/classes/Versions.php

```diff
@@ -568,8 +568,25 @@ public function compare($blnReturnBuffer=false)
 						// Decode entities if the "decodeEntities" flag is not set (see #360)
 						if (empty($GLOBALS['TL_DCA'][$this->strTable]['fields'][$k]['eval']['decodeEntities']))
 						{
-							$to[$k] = StringUtil::decodeEntities($to[$k]);
-							$from[$k] = StringUtil::decodeEntities($from[$k]);
+							if (\is_array($to[$k])) {
+								array_walk_recursive($to[$k], static function (&$value) {
+									$value = StringUtil::decodeEntities($value);
+								});
+							}
+							else
+							{
+								$to[$k] = StringUtil::decodeEntities($to[$k]);
+							}
+
+							if (\is_array($from[$k])) {
+								array_walk_recursive($from[$k], static function (&$value) {
+									$value = StringUtil::decodeEntities($value);
+								});
+							}
+							else
+							{
+								$from[$k] = StringUtil::decodeEntities($from[$k]);
+							}
```

Comment:
```suggestion
							$to[$k] = \is_string($to[$k]) ? StringUtil::decodeEntities($to[$k]) : $to[$k];
							$from[$k] = \is_string($from[$k]) ? StringUtil::decodeEntities($from[$k]) : $from[$k];
```

Wouldn’t this be a more appropriate fix as we cannot reproduce where the array is coming from?

Reply:
Not sure what you mean. It feels perfectly valid to be an array value in a diff field. As far as I know, the diff field also supports that. And it should also decode entities on these values?

Reply:
I probably misunderstood something then. 
How can I reproduce your case?

Reply:
You need to get array data into `tl_version.data`. I guess it is easiest if you create a new version record and then manipulate `data` with something like `serialize(['field' => ['foo' => 'bar']])`

Reply:
I think we'll need an actual reproduction. If I have the time I'll try to reproduce this with a dataset that uses the `m-vo/group-widget`.

Reply:
@fritzmg did you find some time by accident? 😅

Reply:
I tried with an example like [this](https://github.com/m-vo/contao-group-widget?tab=readme-ov-file#tldr) where I then saved the content element with various changes multiple times and then restored to a previous versions. But I was not able to reproduce the issue.

Reply:
As discussed in the Contao call, we cannot merge this without a reproduction case.

Reply:
Closing then since I don't really care